core/src/main/java/com/sequenceiq/cloudbreak/service/rdsconfig/RedbeamsDbCertificateProvider.java

```diff
@@ -44,13 +44,17 @@ public String getSslCertsFilePath() {
 
     public RedbeamsDbSslDetails getRelatedSslCerts(StackDto stackDto) {
         Set<String> relatedSslCerts = new HashSet<>();
+        Set<String> sslCertsForStack = new HashSet<>();
         getDatalakeDatabaseRootCerts(stackDto.getStack(), relatedSslCerts);
         ClusterView cluster = stackDto.getCluster();
-        Set<String> sslCertsForStack = getDatabaseRootCerts(cluster.getName(), cluster.getDatabaseServerCrn(), stackDto.getStack().getResourceCrn());
-        relatedSslCerts.addAll(sslCertsForStack);
-        //TODO persist the gathered certs for the cluster to support cert rotation in the future
+        if (dbServerConfigurer.isRemoteDatabaseRequested(cluster.getDatabaseServerCrn())) {
+            sslCertsForStack = getDatabaseRootCerts(cluster.getName(), cluster.getDatabaseServerCrn(), stackDto.getStack().getResourceCrn());
+            relatedSslCerts.addAll(sslCertsForStack);
+            //TODO persist the gathered certs for the cluster to support cert rotation in the future
 
-        // Note: When stackDto is a DH, relatedSslCerts is the union of DL + DH certs, and sslEnabledForStack is purely determined by the presence of DH certs.
+            // Note: When stackDto is a DH, relatedSslCerts is the union of DL + DH certs,
+            // and sslEnabledForStack is purely determined by the presence of DH certs.
+        }
         return new RedbeamsDbSslDetails(relatedSslCerts, !sslCertsForStack.isEmpty());
```

Comment:
The comment was supposed to refer to the `return` statement, so I suggest moved it right after the `if` block.